File: .claude/battle-test/test_ui.py
#!/usr/bin/env python3
"""
UI Testing Script for Mason Dashboard
Tests each page for rendering, console errors, and basic functionality.
"""

from playwright.sync_api import sync_playwright
import json
import time
from datetime import datetime

BASE_URL = "http://localhost:3000"

PAGES = [
    {"path": "/", "name": "landing"},
    {"path": "/auth/signin", "name": "signin"},
    {"path": "/setup", "name": "setup"},
    {"path": "/admin/backlog", "name": "backlog"},
    {"path": "/settings/database", "name": "database-settings"},
    {"path": "/settings/github", "name": "github-settings"},
    {"path": "/settings/api-keys", "name": "api-keys"},
]

def test_page(page, page_info):
    """Test a single page and return results."""
    url = BASE_URL + page_info["path"]
    name = page_info["name"]

    result = {
        "url": page_info["path"],
        "status": "pass",
        "load_time_ms": 0,
        "console_errors": [],
        "issues": []
    }

    # Collect console errors
    console_errors = []
    def handle_console(msg):
        if msg.type == "error":
            console_errors.append(msg.text)

    page.on("console", handle_console)

    try:
        # Navigate and measure load time
        start_time = time.time()
        page.goto(url, timeout=15000)
        page.wait_for_load_state('networkidle', timeout=15000)
        load_time = int((time.time() - start_time) * 1000)
        result["load_time_ms"] = load_time

        # Check if page has content
        body_text = page.locator('body').text_content()

        if not body_text or len(body_text.strip()) < 10:
            result["status"] = "fail"
            result["issues"].append({
                "type": "blank_page",
                "severity": "critical",
                "description": "Page appears to be blank or has minimal content",
                "screenshot": f".claude/battle-test/screenshots/UI-{name}-blank.png"
            })
            page.screenshot(path=f".claude/battle-test/screenshots/UI-{name}-blank.png", full_page=True)

        # Check for error text on page
        error_patterns = ["Error", "Something went wrong", "Unexpected", "Failed to", "Cannot"]
        for pattern in error_patterns:
            try:
                error_elem = page.locator(f"text=/{pattern}/i").first
                if error_elem.is_visible(timeout=1000):
                    error_text = error_elem.text_content()
                    result["status"] = "fail"
                    result["issues"].append({
                        "type": "render_fail",
                        "severity": "high",
                        "description": f"Error text found on page: {error_text}",
                        "screenshot": f".claude/battle-test/screenshots/UI-{name}-error.png"
                    })
                    page.screenshot(path=f".claude/battle-test/screenshots/UI-{name}-error.png", full_page=True)
                    break
            except:
                pass  # Pattern not found, continue

        # Check console errors
        if console_errors:
            result["console_errors"] = console_errors
            # Filter out noise and determine if critical
            critical_errors = [e for e in console_errors if not any(noise in e.lower() for noise in ["favicon", "net::err_failed"])]

            if critical_errors:
                result["status"] = "fail"
                result["issues"].append({
                    "type": "console_error",
                    "severity": "medium",
                    "description": f"Console errors detected: {len(critical_errors)} errors",
                    "screenshot": f".claude/battle-test/screenshots/UI-{name}-console.png"
                })
                page.screenshot(path=f".claude/battle-test/screenshots/UI-{name}-console.png", full_page=True)

    except Exception as e:
        result["status"] = "fail"
        result["issues"].append({
            "type": "timeout",
            "severity": "critical",
            "description": f"Failed to load page: {str(e)}",
            "screenshot": f".claude/battle-test/screenshots/UI-{name}-timeout.png"
        })
        try:
            page.screenshot(path=f".claude/battle-test/screenshots/UI-{name}-timeout.png", full_page=True)
        except:
            pass

    return result

def main():
    """Run all UI tests and generate report."""
    results = {
        "agent_id": "UI-1",
        "completed_at": datetime.utcnow().isoformat() + "Z",
        "summary": {
            "pages_tested": 0,
            "pages_passed": 0,
            "pages_failed": 0,
            "issues_found": 0
        },
        "pages": []
    }

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        for page_info in PAGES:
            print(f"Testing {page_info['path']}...")
            result = test_page(page, page_info)
            results["pages"].append(result)

            results["summary"]["pages_tested"] += 1
            if result["status"] == "pass":
                results["summary"]["pages_passed"] += 1
                print(f"  ✓ PASS ({result['load_time_ms']}ms)")
            else:
                results["summary"]["pages_failed"] += 1
                results["summary"]["issues_found"] += len(result["issues"])
                print(f"  ✗ FAIL - {len(result['issues'])} issue(s)")
                for issue in result["issues"]:
                    print(f"    - {issue['type']}: {issue['description']}")

        browser.close()

    # Write results
    output_path = ".claude/battle-test/results/UI-1.json"
    with open(output_path, 'w') as f:
        json.dump(results, f, indent=2)

    print(f"\n{'='*60}")
    print(f"UI Testing Complete")
    print(f"{'='*60}")
    print(f"Pages Tested: {results['summary']['pages_tested']}")
    print(f"Pages Passed: {results['summary']['pages_passed']}")
    print(f"Pages Failed: {results['summary']['pages_failed']}")
    print(f"Issues Found: {results['summary']['issues_found']}")
    print(f"\nResults written to: {output_path}")

if __name__ == "__main__":
    main()

File: .claude/battle-test/test-ui.js
#!/usr/bin/env node
/**
 * UI Testing Script for Mason Dashboard
 * Tests each page for rendering, console errors, and basic functionality.
 */

import { chromium } from 'playwright';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const BASE_URL = 'http://localhost:3000';

const PAGES = [
  { path: '/', name: 'landing' },
  { path: '/auth/signin', name: 'signin' },
  { path: '/setup', name: 'setup' },
  { path: '/admin/backlog', name: 'backlog' },
  { path: '/settings/database', name: 'database-settings' },
  { path: '/settings/github', name: 'github-settings' },
  { path: '/settings/api-keys', name: 'api-keys' },
];

async function testPage(page, pageInfo) {
  const url = BASE_URL + pageInfo.path;
  const name = pageInfo.name;

  const result = {
    url: pageInfo.path,
    status: 'pass',
    load_time_ms: 0,
    console_errors: [],
    issues: [],
  };

  // Collect console errors
  const consoleErrors = [];
  page.on('console', (msg) => {
    if (msg.type() === 'error') {
      consoleErrors.push(msg.text());
    }
  });

  try {
    // Navigate and measure load time
    const startTime = Date.now();
    await page.goto(url, { timeout: 15000 });
    await page.waitForLoadState('networkidle', { timeout: 15000 });
    const loadTime = Date.now() - startTime;
    result.load_time_ms = loadTime;

    // Check if page has content
    const bodyText = await page.locator('body').textContent();

    if (!bodyText || bodyText.trim().length < 10) {
      result.status = 'fail';
      result.issues.push({
        type: 'blank_page',
        severity: 'critical',
        description: 'Page appears to be blank or has minimal content',
        screenshot: `.claude/battle-test/screenshots/UI-${name}-blank.png`,
      });
      await page.screenshot({
        path: `.claude/battle-test/screenshots/UI-${name}-blank.png`,
        fullPage: true,
      });
    }

    // Check for error text on page
    const errorPatterns = [
      'Error',
      'Something went wrong',
      'Unexpected',
      'Failed to',
      'Cannot',
    ];
    for (const pattern of errorPatterns) {
      try {
        const errorElem = page.locator(`text=/${pattern}/i`).first();
        const isVisible = await errorElem
          .isVisible({ timeout: 1000 })
          .catch(() => false);
        if (isVisible) {
          const errorText = await errorElem.textContent();
          result.status = 'fail';
          result.issues.push({
            type: 'render_fail',
            severity: 'high',
            description: `Error text found on page: ${errorText}`,
            screenshot: `.claude/battle-test/screenshots/UI-${name}-error.png`,
          });
          await page.screenshot({
            path: `.claude/battle-test/screenshots/UI-${name}-error.png`,
            fullPage: true,
          });
          break;
        }
      } catch (err) {
        // Pattern not found, continue
      }
    }

    // Check console errors
    if (consoleErrors.length > 0) {
      result.console_errors = consoleErrors;
      // Filter out noise
      const criticalErrors = consoleErrors.filter(
        (e) =>
          !e.toLowerCase().includes('favicon') &&
          !e.toLowerCase().includes('net::err_failed'),
      );

      if (criticalErrors.length > 0) {
        result.status = 'fail';
        result.issues.push({
          type: 'console_error',
          severity: 'medium',
          description: `Console errors detected: ${criticalErrors.length} error(s)`,
          screenshot: `.claude/battle-test/screenshots/UI-${name}-console.png`,
        });
        await page.screenshot({
          path: `.claude/battle-test/screenshots/UI-${name}-console.png`,
          fullPage: true,
        });
      }
    }
  } catch (error) {
    result.status = 'fail';
    result.issues.push({
      type: 'timeout',
      severity: 'critical',
      description: `Failed to load page: ${error.message}`,
      screenshot: `.claude/battle-test/screenshots/UI-${name}-timeout.png`,
    });
    try {
      await page.screenshot({
        path: `.claude/battle-test/screenshots/UI-${name}-timeout.png`,
        fullPage: true,
      });
    } catch (screenshotErr) {
      // Ignore screenshot errors
    }
  }

  return result;
}

async function main() {
  const results = {
    agent_id: 'UI-1',
    completed_at: new Date().toISOString(),
    summary: {
      pages_tested: 0,
      pages_passed: 0,
      pages_failed: 0,
      issues_found: 0,
    },
    pages: [],
  };

  // Ensure screenshot directory exists
  const screenshotDir = path.join(__dirname, 'screenshots');
  if (!fs.existsSync(screenshotDir)) {
    fs.mkdirSync(screenshotDir, { recursive: true });
  }

  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  for (const pageInfo of PAGES) {
    console.log(`Testing ${pageInfo.path}...`);
    const result = await testPage(page, pageInfo);
    results.pages.push(result);

    results.summary.pages_tested += 1;
    if (result.status === 'pass') {
      results.summary.pages_passed += 1;
      console.log(`  ✓ PASS (${result.load_time_ms}ms)`);
    } else {
      results.summary.pages_failed += 1;
      results.summary.issues_found += result.issues.length;
      console.log(`  ✗ FAIL - ${result.issues.length} issue(s)`);
      for (const issue of result.issues) {
        console.log(`    - ${issue.type}: ${issue.description}`);
      }
    }
  }

  await browser.close();

  // Ensure results directory exists
  const resultsDir = path.join(__dirname, 'results');
  if (!fs.existsSync(resultsDir)) {
    fs.mkdirSync(resultsDir, { recursive: true });
  }

  // Write results
  const outputPath = path.join(resultsDir, 'UI-1.json');
  fs.writeFileSync(outputPath, JSON.stringify(results, null, 2));

  console.log('\n' + '='.repeat(60));
  console.log('UI Testing Complete');
  console.log('='.repeat(60));
  console.log(`Pages Tested: ${results.summary.pages_tested}`);
  console.log(`Pages Passed: ${results.summary.pages_passed}`);
  console.log(`Pages Failed: ${results.summary.pages_failed}`);
  console.log(`Issues Found: ${results.summary.issues_found}`);
  console.log(`\nResults written to: ${outputPath}`);

  // Exit with error code if any tests failed
  process.exit(results.summary.pages_failed > 0 ? 1 : 0);
}

main().catch((error) => {
  console.error('Fatal error:', error);
  process.exit(1);
});
